### ISSUE TITLE
RTSP streams freeze at version 3.1+ but not at 3.0

### ISSUE THREAD

**Issue by miloMich** [opencv/opencv#11619]
##### System information (version)

- OpenCV => 3.1+ including 3.4.1
- Operating System / Platform => Windows .Net C# x86
- Compiler => Visual Studio 2017

##### Detailed description

Using an RTSP camera such as "AimCam", the stream freezes after 10 seconds. It does not freeze when using EMGU 3.0.  It does not freeze when run through VLC utility either

**Comment by alalek** [comment]
> EMGU

There is no support for C# bindings here.

> incomplete

It is very hard to reproduce issues with remote video streams.
You should investigate this problem yourself and provide more details.

**Comment by miloMich** [comment]
Regardless of binding, it appears the problem is directly within the OpenCV library though. It is verified to not exist in version 3, yet does exist in later versions.

**Comment by kinchungwong** [comment]
Please try to explain "freeze". Please try to describe the symptoms with more technicality. Unfortunately, the issues page is not a place to solicit for troubleshooting assistance. While there is answers.opencv.org  which allows open-ended questions to be asked, there is definitely a need for a place where experienced troubleshooters and symptom observers can collaborate.

Have you tried running all of the processes with the debugger attached?

**Comment by berak** [comment]
hmm, you're closing the rtsp stream, and reopening a (local) usb webcam stream here:
```
int apiID = cv::CAP_ANY;      // 0 = autodetect default API
// open selected camera using selected API
    cap.open(deviceID + apiID);
```
(that's for sure not, what you wanted, also not helpful to debug the original problem)

**Comment by miloMich** [comment]
agreed - that line was left there by mistake when copied to this thread

**Comment by berak** [comment]
i'll strike the comment, you update your code above, i'd say.

**Comment by miloMich** [comment]
The request for a different URL is understood. Unfortunately, this is a URL relevant to the "AimCam" camera I have issues with. When testing against the public "bunny" video at the link below, has no issue. When running through debugger - the application hands within the OpenCV_ffmpeg dll.

rtsp://184.72.239.149/vod/mp4:BigBuckBunny_175k.mov

the aim-cam camera model: (this is the store page, no intention to solicit purchase)
https://www.aimcam.com/products/aimcam-pro-2-black

like mentioned above. The steam does not halt process when run with version 3.0 of open-cv. Also, the stream does not "freeze" (as explained above) when run through VLC or through its own android application.

**Comment by berak** [comment]
rtsp://184.72.239.149/vod/mp4:BigBuckBunny_175k.mov -- invalid, sorry to say so.

**Comment by berak** [comment]
3.4.0 -- no more relevant. please use at least 3.4.1 , better even. latest master branch (4.0.0.dev)

**Comment by berak** [comment]
no problem, using latest master branch here: 

```
int main(int argc, char** argv)
{
	VideoCapture cap("rtsp://184.72.239.149/vod/mp4:BigBuckBunny_175k.mov");
	cout << cap.isOpened() << endl;
	while(1) {
		Mat frame;
		cap >> frame;
		imshow("M",frame);
		waitKey(10);
	}
    return 0;
}
```

( emgu or not, i don't think, you have a valid issue here, in the 1st case)

**Comment by miloMich** [comment]
Like I mentioned, the "bunny" video isn't demonstrating the issue. I can only reproduce this issue with the camera I mention above, which is streaming on its own wifi and cannot be made public. And it appears to be locked at the ffmpeg dll. I'm in the process of building it again with 3.4.1 to verify again.

**Comment by berak** [comment]
@miloMich apologies for being overly harsh before !

(but again, -- as long as we cannot access your ipcam, we can't debug your problem)

**Comment by miloMich** [comment]
Understood - I'll see if anything further can be done to assist. If it's OK - I'll leave this open for the next few days as I look into ways to provide a sample stream

**Comment by kinchungwong** [comment]
On the internet there are instructions for saving the stream to a file (without any re-encoding). Please give it a try, and see if the freeze problem can be re-created when reading from that file.

Based on internet discussions, it is true that ip cameras can behave erratically, such as closing streams unexpectedly or reboot. The only way to investigate would be to ask the ip camera manufacturer to investigate. Otherwise, another venue is to ask in the user forums for that vendor/brand of ip camera.

**Comment by alalek** [comment]
OpenCV doesn't work with cameras / video streams directly. It uses some libraries/SDK/APIs.
In this case OpenCV uses FFmpeg to grab rstp stream.

opencv_ffmpeg.dll is a wrapper over FFmpeg sources.
OpenCV 3.0 uses very old FFmpeg (something like FFmpeg 2.x).
OpenCV 3.4.1 uses FFmpeg 3.4.2.

1.
Did you observe issues with FFmpeg tools without OpenCV? Try "ffmpeg" binary to capture your rstp stream and write it into file.
If possible try both fresh (3.4) and old (2.x) FFmpeg versions.

2.
Try to play with TCP/UDP settings (OpenCV 3.3.1+ via `OPENCV_FFMPEG_CAPTURE_OPTIONS`): https://github.com/opencv/opencv/blob/3.4.1/modules/videoio/src/cap_ffmpeg_impl.hpp#L811-L823

3.
Also you could try to "replace" this wrapper DLL with older one. External interfaces of opencv_ffmpeg wrapper may be compatible (not tested!), so try to rename opencv_ffmpeg DLL from OpenCV 3.0 to 3.4.1.
